tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -81,6 +81,124 @@ sub do_sanity_checks {
     assert_script_run($ssh_vm . "vmware-toolbox-cmd logging level get vmtoolsd | grep 'vmtoolsd.level = debug'");
 }
 
+sub do_power_mgmt_tests {
+    my ($vm_id, $vm_ip) = @_;
+    my ($powerops, $powerops_ret);
+
+    record_info('Power Manangement Tests');
+
+    record_info('Guest Power Shutdown');
+    $powerops = 'power.shutdown';
+    check_vmtools_service() if (is_svirt);
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 0, $VM_POWER_OFF);
+
+    record_info('Guest Power On');
+    $powerops = 'power.on';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_OFF);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    record_info('Guest Power Reboot');
+    $powerops = 'power.reboot';
+    check_vmtools_service() if (is_svirt);
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    record_info('Guest Power Reset');
+    $powerops = 'power.reset';
+    check_vmtools_service() if (is_svirt);
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    record_info('Guest Power Off');
+    $powerops = 'power.off';
+    check_vmtools_service() if (is_svirt);
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 0, $VM_POWER_OFF);
+
+    # Boot up the VM if it's powered off
+    $powerops = 'power.on';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_OFF);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+}
+
+sub do_networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    record_info('Networking Tests');
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    # Test network via assigned IPv4 address
+    if (is_sle('15+')) {
+        assert_script_run($ssh_vm . "ping -I $vm_ip -4 -c3 openqa.suse.de");
```

Comment:
Suggest to replace openqa.suse.de with OPENQA_URL var

Reply:
OK, updated.